### ISSUE TITLE
@entangle.defer batching not working correctly.

### ISSUE THREAD

**Issue by joshhanley** [livewire/livewire#1874]
### Description

On an alpine dropdown where the dropdown state needs to be linked with livewire, using @entangle.defer is dropping some state changes when a request is finally sent back to livewire.

### Exact steps to reproduce

I have created a demo repo here: https://github.com/joshhanley/EntangleDeferBug

More info in the repo

1. Pull down the repo
2. Composer install
3. Open browser and go to site, its in the base route

I have just replaced the contents of welcome.blade.php and added a Livewire component called Dropdown

**Screenshot**

![EntangleDeferBug](https://user-images.githubusercontent.com/882837/96800841-09401300-1452-11eb-8105-87a945547188.gif)

### Stripped-down, copy-pastable code snippets

**Context** (see demo repo)
- Livewire version: 2.3.0
- Laravel version: 8.11.2
- Alpine version: 2.x.x (from CDN, so I think 2.7.1)
- Browser: Chrome

**Comment by uteq** [comment]
My best guess is that this is Alpine related.
Could you try to use x-on.blur in stead of x-on:click.away:

```blade
<input
    x-on:focus="alpineShow = true"
    x-on:blur="alpineShow = false; $wire.someMethod()"
    class=" w-64 p-4 border border-gray-500 rounded bg-white"
    placeholder="Click here"
/>
```

**Comment by joshhanley** [comment]
@uteq Thanks for the suggestion, I've tried what you suggested, but that makes it worse as it never actually closes. 

But you have highlighted an issue with the contrived example (theres always something when you strip it back to a bare bones problem haha).

The click.away should actually be on the wrapping div, so the user can interact with the input and the dropdown, and it closes when anywhere outside of it is clicked.

I've now moved the click.away to the wrapping div and it has changed the issue slightly, the dropdown is closing but with it flashing back open in the middle. It is still sending two network requests though, where as i thought it would only send one.

![image](https://user-images.githubusercontent.com/882837/96851448-c0b14580-14a3-11eb-94a2-a58dbe09d7e1.png)
![image](https://user-images.githubusercontent.com/882837/96851478-c73fbd00-14a3-11eb-8983-359c0ede8203.png)
